woodwork/schema.py

```diff
@@ -154,27 +154,46 @@ def time_index(self):
                 return col_name
         return None
 
-    def set_types(self, semantic_tags=None, retain_index_tags=True):
-        """Update the semantic tags for any columns names in the provided types dictionary,
-        updating the Woodwork typing information for the DataFrame.
+    def set_types(self, logical_types=None, semantic_tags=None, retain_index_tags=True):
+        """Update the logical tpye and semantic tags for any columns names in the provided types dictionaries,
+        updating the Schema at those columns.
 
         Args:
+            logical_types (dict[str -> LogicalType], optional): A dictionary defining the new logical types for the
```

Comment:
I don't know that we want the `Schema.set_types` to only take in LogicalType objects. It follows the way that the Schema only takes in LogicalType objects, but maybe that's less important here.

Reply:
Even in the `Schema.__init__`?

I agree that we want to avoid people having to import more LogicalTypes than they have to, but, in theory, they should rarely be directly creating Schemas or calling Schema.set_types, so it probably won't be that big of an issue.

then again, I can't think of a reason that a Schema shouldn't be able to parse logical type strings other than trying to simplify it a bit.

Reply:
Sorry - I'm getting confused between the methods. I think it is fine for the Schema methods to only take in LogicalType classes, but the accessor methods should be more flexible.

Reply:
Ahh, got it. So to confirm: We'll allow string inputs in `WoodworkTableAccessor.set_types` but not for `Schema.set_types`

Reply:
Yep, that's what I was thinking.